src/cron/checkReferralReminder.ts

```diff
@@ -0,0 +1,25 @@
+import { Cron } from './cron';
+import { Alerts } from '../entity';
+
+const cron: Cron = {
+  name: 'generic-referral-reminder',
+  handler: async (con, logger) => {
+    logger.info('updating referral reminders...');
+    const { affected } = await con
+      .getRepository(Alerts)
+      .createQueryBuilder()
+      .update()
+      .where({ showGenericReferral: false })
+      .andWhere(
+        `(
+              dateCastIndex(flags, 'lastReferralReminder') <= NOW() - INTERVAL '6 months'
+          OR  (dateCastIndex(flags, 'lastReferralReminder') IS NULL AND (SELECT u."createdAt" FROM public.user AS u WHERE u.id = "userId") <= NOW() - INTERVAL '2 weeks')
+        )`,
```

Comment:
Due to the nature of the complexity to fetch the data, have resorted to manual query and subquery for the user as it is [not available through typeorm](https://github.com/typeorm/typeorm/issues/564#issuecomment-310331468)

Reply:
Not sure this supports, but could we not do:
- where reminder is null AND createdAt 2 weeks ago
- OR where reminder is 6 months ago?

Maybe it's the subquery efficiency.
If not this is fine from my side to be honest.

Reply:
Previously, we can't since joining was not possible but since we will introduce a trigger on user creation for alerts, we can probably use the `createdAt` from Alerts entity.

Reply:
After checking, it seems there is no `createdAt` in Alerts table, meaning, we will have to stick with what's introduced here.